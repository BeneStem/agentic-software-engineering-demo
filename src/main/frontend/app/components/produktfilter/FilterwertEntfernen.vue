<template>
  <button type="button" role="tab" @click="filterwertEntfernen" class="erk_finden__filterwert-entfernen">
    {{ filterwertName }}
    <B2kIcon name="Schliessen" class="erk_finden__filterwert-entfernen-schliessen" />
  </button>
</template>

<script setup lang="ts">
import B2kIcon from '@blume2000/design-system/dist/komponenten/atome/B2kIcon.vue';

defineProps<{
  filterwertName: string;
}>();

const emit = defineEmits(['filterwertEntfernen']);

function filterwertEntfernen() {
  emit('filterwertEntfernen');
}
</script>

<style lang="scss" scoped>
button {
  display: flex;
  align-items: center;
  @include font-Rubik-light(16);
}
.erk_finden__filterwert-entfernen {
  display: flex;

  &-schliessen {
    margin-left: 4px;
    width: 16px;
    height: 16px;
  }
}
</style>
