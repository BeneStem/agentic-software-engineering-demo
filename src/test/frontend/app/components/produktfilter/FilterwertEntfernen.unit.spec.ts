import { mount } from '@vue/test-utils';
import FilterwertEntfernen from '@/main/frontend/app/components/produktfilter/FilterwertEntfernen.vue';

describe('FilterwertEntfernen.vue', () => {
  it('Rendert FilterwertEntfernen', () => {
    // Given
    const wrapper = mount(FilterwertEntfernen, {
      props: {
        filterwertName: 'FilterwertName',
      },
    });

    // Then
    expect(wrapper.element).toMatchSnapshot();
  });
});
