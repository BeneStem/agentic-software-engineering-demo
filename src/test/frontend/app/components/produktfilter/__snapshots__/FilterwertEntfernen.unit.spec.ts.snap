// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterwertEntfernen.vue Rendert FilterwertEntfernen 1`] = `
<button
  class="erk_finden__filterwert-entfernen"
  role="tab"
  type="button"
>
  FilterwertName 
  <svg
    aria-hidden="true"
    class="erk_finden__filterwert-entfernen-schliessen"
    role="img"
  >
    <!--v-if-->
    <use
      href="/static/assets/svgs/icons/Schliessen.svg#svg"
    />
  </svg>
</button>
`;
